### ISSUE TITLE
Common up trlbuf.c into hstcal/lib

### ISSUE THREAD

**Issue by jamienoss** [spacetelescope/hstcal#191]
### Update:
This is dependent on #207 

__Note to self:__ Remeber to change this (as done in #194)

```c
void trlmessage (char *message) {
  
      void asnmessage (char *);
  
 -    if (!message)
 +    if (!message || !*message)
          return;
```
### UPDATE:
Done.

``trlbuf`` should also not be global as it is [here](https://github.com/spacetelescope/hstcal/blob/master/pkg/wfc3/calwf3/lib/trlbuf.c#L127).

### UPDATE:
This was moved into trlbuf.c as the following global:
```c
struct TrlBuf trlbuf = { 0 };
```

Should this be global at all?

### Update:
Done. Converted to: ``extern struct TrlBuf trlbuf ;`` and then declared within each  ``main()`` as needed.

May have to declare as extern in a few (all) places. Already done in trlbuf.c (this might be all).

### Update:
Change ``trlbuf.c::CloseTrlBuf(void)`` to reflect NULL changes made from #196 
```c
if (trlbuf.buffer)
        {
            free (trlbuf.buffer);
            trlbuf.buffer = NULL;
        }

        if (trlbuf.preface)
        {
            free (trlbuf.preface);
            trlbuf.preface = NULL;
        }
```

### Update:
Done.

**Comment by jamienoss** [comment]
__Note to self:__ Remeber to change this (as done in #194)

```c
void trlmessage (char *message) {
  
      void asnmessage (char *);
  
 -    if (!message)
 +    if (!message || !*message)
          return;
```

**Comment by jamienoss** [comment]
``trlbuf`` should also not be global as it is [here](https://github.com/spacetelescope/hstcal/blob/master/pkg/wfc3/calwf3/lib/trlbuf.c#L127).

**Comment by jamienoss** [comment]
Make sure that extra bits pulled out into hstcal.h are true for stis also - currently (wip code, not master) they are not.

**Comment by jamienoss** [comment]
Should trl_prefix be the same for both calacs and calwf3, does it mater?

**Comment by jamienoss** [comment]
This has to pull out common decs from acs.h, wf3.h, & msg.h that are needed in the common trl.c. These will go into include/hstcal.h.

**Comment by jamienoss** [comment]
See if some of the includes in ctegen2 to acs & wf3 can be removed?

**Comment by jamienoss** [comment]
Double check all decs added to hstcal.h exist nowhere else.

**Comment by jamienoss** [comment]
The work in #231 may still not be considered an adequate/elegant refactor. The basis for this is that two ``extern`` objects exist ``MsgText`` and ``trlBuf`` that are defined within the ``main(void)`` code however, they are referenced from within a library. This prevents the library ``hstcallib`` that this code now belongs to, from really being a library. These two objects should not be global at all but passed around instead.
